packages/projection-typeorm/src/operators/storeHandles.ts

```diff
@@ -0,0 +1,38 @@
+import { ChainSyncEventType } from '@cardano-sdk/core';
+import { HandleEntity } from '../entity';
+import { Mappers } from '@cardano-sdk/projection';
+import { typeormOperator } from './util';
+
+export const storeHandles = typeormOperator<Mappers.WithHandles & Mappers.WithMint>(
+  async ({ mint: _mint, handles: _handles, queryRunner, eventType }) => {
+    try {
+      if (_handles.length === 0) return;
+
+      const handleRepository = queryRunner.manager.getRepository(HandleEntity);
+      const existingHandles = await handleRepository.find();
```

Comment:
This sounds to make a full table scan on handles at each event, even if there are no handles to _handle_ in an event.
Isn't better to find for a specific handle right before checking if we need to insert or update a record?

Reply:
```suggestion
      const existingHandles = await handleRepository.find(handles);
```
But yes you're right @iccicci, something like this would be the expected implementation here. The repository is responsible for accessing the records we need.